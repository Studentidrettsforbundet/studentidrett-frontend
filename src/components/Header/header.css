.header {
    padding: 1rem;
    background-color: #ffab40;
    width: 100vw !important;
    max-width: 100% !important;
}
.headerContainer {
    display: flex;
    justify-content: space-between;
}
.headerLink {
    width: fit-content;
}

.headerContent {
    display: flex;
    flex-direction: row;
    align-items: center;
}

img {
    height: 53px;
}

.textContent {
    margin-left: 1rem;
    font-size: 16px;
    line-height: 20px;
    font-family: ko355, sans-serif;
}
