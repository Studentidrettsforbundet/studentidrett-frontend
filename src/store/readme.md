# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
