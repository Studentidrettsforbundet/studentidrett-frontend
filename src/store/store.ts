import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
