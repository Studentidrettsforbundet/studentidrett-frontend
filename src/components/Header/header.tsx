import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import './header.css';
import SearchIcon from '../SearchBar/searchIcon';
import '../SearchBar/styles.css';
import { useSelector } from 'react-redux';
import { combinedStateInterface } from '../../store/store';

const Header = (): JSX.Element => {
    const location = useLocation();
    const reduxState = useSelector((state: combinedStateInterface) => state);
    const pathnameRegex = /\bquestionnaire\b|\bresult\b|\bsearch\b/;
    const isLandingOrQuestionnaire = location.pathname.match(pathnameRegex) || location.pathname === '/';

    return (
        <div className="container headerContainer">
            <section className="headerLink">
                <Link to="/" style={{ textDecoration: 'none', color: 'black' }}>
                    <section className="headerContent">
                        <img src={require('../../assets/logo.png')} alt="logo" />
                        <div className="textContent">
                            Norges
                            <br />
                            Studentidrettsforbund
                        </div>
                    </section>
                </Link>
            </section>
            {!isLandingOrQuestionnaire ? (
                <div className="search_icon-container">
                    <SearchIcon {...{ currentState: reduxState.searchBar.showSearchBar }} />
                </div>
            ) : (
                <Link to={'/search'} className="search_icon-container">
                    <SearchIcon {...{ currentState: false }} />
                </Link>
            )}
        </div>
    );
};

export default Header;
