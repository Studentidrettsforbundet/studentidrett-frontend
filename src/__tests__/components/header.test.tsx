import React from 'react';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../../components/Header/header';
import { createMemoryHistory } from 'history';
import store from '../../store/store';
import { Provider } from 'react-redux';
const history = createMemoryHistory();

describe('header', () => {
    describe('snapshot tests', () => {
        it('should render correctly', () => {
            const tree = renderer
                .create(
                    <Provider store={store}>
                        <Router history={history}>
                            <Header />
                        </Router>
                    </Provider>,
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
