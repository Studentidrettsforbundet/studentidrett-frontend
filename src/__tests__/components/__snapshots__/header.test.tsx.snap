// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot tests should render correctly 1`] = `
<div
  className="header"
>
  <div
    className="container headerContainer"
  >
    <section
      className="headerLink"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <section
          className="headerContent"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <div
            className="textContent"
          >
            Norges
            <br />
            Studentidrettsforbund
          </div>
        </section>
      </a>
    </section>
    <a
      className="search_icon-container"
      href="/search"
      onClick={[Function]}
    >
      <img
        alt="Search icon"
        className="search_icon"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "search.svg",
          }
        }
      />
    </a>
  </div>
</div>
`;
